UnitTests/BraintreeThreeDSecureTests/BTThreeDSecureClient_Tests.swift

```diff
@@ -281,12 +287,17 @@ class BTThreeDSecureClient_Tests: XCTestCase {
         threeDSecureRequest.threeDSecureRequestDelegate = mockThreeDSecureRequestDelegate
 
         let expectation = expectation(description: "willCallCompletion")
+        mockAPIClient.cannedConfigurationResponseBody = mockConfiguration
+        mockCompletion(["dfReferenceId": "fake-df-reference-id"])
 
-        mockAPIClient.cannedConfigurationResponseBody = BTJSON(value: [
-            "threeDSecure": ["cardinalAuthenticationJWT": "FAKE_JWT"],
-            "assetsUrl": "http://assets.example.com"
-        ] as [String: Any])
-        let client = BTThreeDSecureClient(apiClient: mockAPIClient)
+        let mockThreeDSecureV2Provider = MockThreeDSecureV2Provider(
+            configuration: BTConfiguration(json: mockConfiguration),
+            apiClient: mockAPIClient,
+            request: threeDSecureRequest,
+            completion: mockCompletion
+        )
+
```

Comment:
Is the purpose of this `mockThreeDSecureV2Provider` to avoid having to make `apiClient.fetchOrReturnRemoteConfiguration` call?

Reply:
Good question - it's to prevent[ this call](https://github.com/braintree/braintree_ios/blob/9576e337797013bc2874f8b75f6e21989a21549c/Sources/BraintreeThreeDSecure/BTThreeDSecureClient.swift#L233-L242) (which eventually calls Cardinal's [setup()](https://github.com/braintree/braintree_ios/blob/9576e337797013bc2874f8b75f6e21989a21549c/Sources/BraintreeThreeDSecure/BTThreeDSecureV2Provider.swift#L48-L54). Setup does API calls we want to prevent for UnitTests).

Reply:
Yeah you are right, we are essentially forcing the tests not to care about when `threeDSecureV2Provider = BTThreeDSecureV2Provider()` is called. We can create a `MockThreeDSecureClient` class though and override `prepareLookup` where the above is called to get around this. Just tested this locally, will push the commit up shortly. Good catch!

Reply:
Updated here: https://github.com/braintree/braintree_ios/pull/1067/commits/e5215687f8e211b5696a841770b7122b0fb92d88